app/src/main/java/net/squanchy/eventdetails/domain/view/ExperienceLevel.kt

```diff
@@ -0,0 +1,37 @@
+package net.squanchy.eventdetails.domain.view
+
+import android.support.annotation.ColorRes
+import android.support.annotation.StringRes
+import net.squanchy.R
+import net.squanchy.support.lang.Optional
+import timber.log.Timber
+import java.util.Locale
+
+enum class ExperienceLevel(
+        private val rawLevel: String,
+        @StringRes val labelStringResId: Int,
+        @ColorRes val colorResId: Int
+) {
+    BEGINNER("beginner", R.string.experience_level_beginner, R.color.experience_level_beginner),
+    INTERMEDIATE("intermediate", R.string.experience_level_intermediate, R.color.experience_level_intermediate),
+    ADVANCED("advanced", R.string.experience_level_advanced, R.color.experience_level_advanced);
+
+    companion object {
+
+        fun fromNullableRawLevel(rawLevel: String): Optional<ExperienceLevel> =
+                try {
+                    Optional.fromNullable(rawLevel).map({ fromRawLevel(it) })
```

Comment:
`fromNullable` doesn't really make sense here since `rawLevel` is always not null, we might want to refactor this part a little bit. If it's actually nullable, we could do something like `rawLevel?.let { fromRawLevel(it) }`. If it's not null, then we can just `fromRawLevel(rawLevel)`

Reply:
Yeah slipped through when converting, nice spot!

Reply:
fd4d09a